src/smt/smt_engine.cpp

```diff
@@ -280,11 +285,24 @@ Result SmtEngine::getStatusOfLastCommand() const
 {
   return d_state->getStatus();
 }
-context::UserContext* SmtEngine::getUserContext()
+context::UserContext* SmtEngine::getUserContext() const
 {
   return d_state->getUserContext();
 }
-context::Context* SmtEngine::getContext() { return d_state->getContext(); }
+context::Context* SmtEngine::getContext() const
+{
+  return d_state->getContext();
+}
+
+TheoryEngine* SmtEngine::getTheoryEngine() const
+{
+  return d_smtSolver->getTheoryEngine();
+}
+
+prop::PropEngine* SmtEngine::getPropEngine() const
+{
+  return d_smtSolver->getPropEngine();
+}
```

Comment:
ditto.

Reply:
Is this about the const?  This is actually required to make things compile in several places.